crates/ruff_python_formatter/src/pattern/pattern_match_mapping.rs

```diff
@@ -1,19 +1,197 @@
+        let PatternMatchMapping {
+            keys,
+            patterns,
+            rest,
+            range: _,
+        } = item;
+
+        debug_assert_eq!(keys.len(), patterns.len());
+
+        let comments = f.context().comments().clone();
+        let dangling = comments.dangling(item);
+
+        if keys.is_empty() && rest.is_none() {
+            return empty_parenthesized("{", dangling, "}").fmt(f);
+        }
+
+        // This node supports three kinds of dangling comments. Most of the complexity originates
+        // with the rest pattern (`{**rest}`), since we can have comments around the `**`, but
+        // also, the `**rest` itself is not a node (it's an identifier), so comments that trail it
+        // are _also_ dangling.
+        //
+        // Specifically, we have these three sources of dangling comments:
+        // ```python
+        // {  # "open parenthesis comment"
+        //    key: pattern,
+        //    **  # end-of-line "double star comment"
+        //    # own-line "double star comment"
+        //    rest  # end-of-line "after rest comment"
+        //    # own-line "after rest comment"
+        // }
+        // ```
+        let (open_parenthesis_comments, double_star_comments, after_rest_comments) =
+            find_double_star(item, f.context().source()).map_or(
+                (dangling, dangling, dangling),
+                |(double_star, rest)| {
+                    let (open_parenthesis_comments, dangling) =
+                        dangling.split_at(dangling.partition_point(|comment| {
+                            comment.line_position().is_end_of_line()
+                                && comment.start() < double_star.start()
+                        }));
+                    let (double_star_comments, after_rest_comments) = dangling.split_at(
+                        dangling.partition_point(|comment| comment.start() < rest.start()),
+                    );
+                    (
+                        open_parenthesis_comments,
+                        double_star_comments,
+                        after_rest_comments,
+                    )
+                },
+            );
+
+        let format_pairs = format_with(|f| {
+            let mut joiner = f.join_comma_separated(item.end());
+
+            for (key, pattern) in keys.iter().zip(patterns) {
+                let key_pattern_pair = KeyPatternPair { key, pattern };
+                joiner.entry(&key_pattern_pair, &key_pattern_pair);
+            }
+
+            if let Some(identifier) = rest {
+                let rest_pattern = RestPattern {
+                    identifier,
+                    comments: double_star_comments,
+                };
+                joiner.entry(&rest_pattern, &rest_pattern);
+            }
+
+            joiner.finish()?;
+
+            trailing_comments(after_rest_comments).fmt(f)?;
+
+            Ok(())
```

Comment:
NIt
```suggestion
            trailing_comments(after_rest_comments).fmt(f)
```

Reply:
I find that I tend to write it this way because it reduces diffs. E.g., it's odd that the _last_ format call omits `?;` "just because" it's the last, unlike the others (reminds me of trailing commas). But I will change :)